engine/robot.py

```diff
@@ -306,29 +183,27 @@ def turn_to_target_heading(self, target_heading, allowed_heading_error, database
                 place.
         """
 
-        predicted_state = self.state  # this will come from Kalman Filter
+        predicted_state = self.robot_state.state  # this will come from Kalman Filter
 
         abs_heading_error = abs(target_heading - float(predicted_state[2]))
 
         while abs_heading_error > allowed_heading_error:
-            if self.is_sim:
-                self.state[2] = np.random.normal(self.state[2], self.heading_noise)
+            if self.robot_state.is_sim:
+                self.robot_state.state[2] = np.random.normal(self.robot_state.state[2], self.robot_state.heading_noise)
             else:
-                self.state = self.update_ekf_step()
-            theta_error = target_heading - self.state[2]
+                self.robot_state.state = self.update_ekf_step()
+            theta_error = target_heading - self.robot_state.state[2]
             w = self.head_pid.update(theta_error)  # angular velocity
-            _, limited_cmd_w = limit_cmds(0, w, self.max_velocity, self.radius)
+            _, limited_cmd_w = limit_cmds(0, w, self.robot_state.max_velocity, self.robot_state.radius)
 
-            if self.is_sim:
-                self.travel(0, self.time_step * limited_cmd_w)
-            else:
-                self.motor_controller.spin_motors(limited_cmd_w, 0)
```

Comment:
if we're removing this, can we add it to travel?

Reply:
oops! accidental deletion

Reply:
added back o7